runtime/v2/module.go

```diff
@@ -161,11 +161,10 @@ func SetupAppBuilder(inputs AppInputs) {
 	app.moduleManager.RegisterInterfaces(inputs.InterfaceRegistrar)
 	app.moduleManager.RegisterLegacyAminoCodec(inputs.LegacyAmino)
 
-	// TODO: this is a bit of a hack, but it's the only way to get the store keys into the app
-	// registerStoreKey could instead set this on StoreOptions directly
 	if inputs.StoreOptions != nil {
 		inputs.AppBuilder.storeOptions = inputs.StoreOptions
 		inputs.AppBuilder.storeOptions.StoreKeys = inputs.AppBuilder.app.storeKeys
+		inputs.AppBuilder.storeOptions.StoreKeys = append(inputs.AppBuilder.storeOptions.StoreKeys, "stf")
```

Comment:
we should open an issue on this. This is pretty hacky .

Reply:
This is cherry-picked from here: https://github.com/cosmos/cosmos-sdk/pull/20611#discussion_r1634945099
It related to the discussion we had last Thursday about IAVL being unable to set state without knowing all the actors before hand.